src/server/app.js

```diff
@@ -35,12 +36,31 @@ http.listen(port, () => {
 });
 
 // socket.io event handlers
-function handleCreate(userId) {
+function handleCreate(userId/*: string */) {
+  console.log(`create room received from: ${userId} `)
   roomId = generateId();
   while (rooms[roomId] != null) {
     console.log('id taken, generating new id');
     roomId = generateId();
   }
-  rooms[roomId] = new Room(roomId);
+  room = new Room(roomId);
+  rooms[roomId] = room;
   console.log('new room created:', rooms);
+  console.log('adding player to room')
+  room.addPlayerToRoom(userId)
 }
+
+function handleJoin(userId/*: string */, roomId/*: string */) {
+  // TODO: proper element membership checking
+  if (rooms[roomId]) {
```

Comment:
prefer explicit nullchecking, `rooms[roomId] != null`. either that or `Object.keys(rooms).includes(roomId)`